To solve question #20722

kebenxiaoming opened the pull request:
website:
https://github.com/owncloud/core/issues/20722

kebenxiaoming committed:
Merge pull request #1 from owncloud/master

合并原作者内容

kebenxiaoming committed:
To solve question #20722

https://github.com/owncloud/core/issues/20722

PVince81 commented on apps/files/js/favoritesfilelist.js:
```diff
@@ -64,6 +64,377 @@ $(document).ready(function() {
+				fileData.displayName = fileData.name + '.d' + Math.floor(fileData.mtime / 1000);
+			return OCA.Files.FileList.prototype._renderRow.call(this, fileData, options);
+		        },
+		        _createRow: function(fileData, options) {
```
No need to copy createRow, _renderRow is enough.

Also on line 70 you need to combine the file name with the file id

kebenxiaoming committed:
Merge branch 'master' into master

kebenxiaoming commented:
No,I made a mistake,sorry,Please close this first.

kebenxiaoming committed:
change combine style

change:

70 line:
fileData.displayName = fileData.name + '.d' + Math.floor(fileData.mtime / 1000); 
change to :
fileData.displayName = fileData.name + '.' + fileData.id;

PVince81 commented:
@kebenxiaoming did you copy createRow as is or did you make specific changes to make it work ?

kebenxiaoming commented:
I have override createRow and also maked some changes to make id work.

PVince81 commented:
Ok thanks. I think it's not good to override it completely because it duplicates a lot of code and makes it hard to maintain. Maybe we need another approach.
I'll have a look at this later.

kebenxiaoming commented:
@PVince81 Yes,we should change this more simply.

PVince81 commented:
@kebenxiaoming if you don't mind I'd like to close this PR and revisit the fixing approach when I find some time to handle the issue.

Unless you want to give this another try at simplifiying the code ? Note that we need a statement regarding the license, see https://github.com/owncloud/core/pull/20754#issuecomment-186721538

kebenxiaoming commented:
@PVince81 Sorry,I don't have enough time to do this recently.And I think that if we want to fix this simply,perhaps we need to modify the core javascript  file:"file.js".

PVince81 commented:
@kebenxiaoming this is totally fine. I'll redo the PR with a simpler approach later.
Thanks!
